docs/reference/migration/migrate_8_5.asciidoc

```diff
@@ -20,3 +20,36 @@ coming::[8.5.0]
 There are no breaking changes in {es} 8.5.
 // end::notable-breaking-changes[]
 
+[discrete]
+[[deprecated-8.5]]
+=== Deprecations
+
+The following functionality has been deprecated in {es} 8.5
+and will be removed in a future version.
+While this won't have an immediate impact on your applications,
+we strongly encourage you take the described steps to update your code
+after upgrading to 8.5.
+
+To find out if you are using any deprecated functionality,
+enable <<deprecation-logging, deprecation logging>>.
+
+
+[discrete]
+[[deprecations_85_network_plugins]]
+==== Plugin API deprecations
+
+[[network_plugins_deprecated]]
+Plugins that extend the NetworkPlugin interface are deprecated.
+[%collapsible]
+====
+*Details* +
+Plugins may override funcionality that controls how nodes connect
+with other nodes over TCP/IP. These plugins extend the NetworkPlugin
+interface. In the next major release, these plugins will fail
+to install.
+
+*Impact* +
+Discontinue using any plugins which extend NetworkPlugin. You can
+see if any plugins use deprecated functionality by checking
+the Elasticsearch deprecation log.
```

Comment:
Additionally, and optionally, we could mention that a deprecated warning will be raised when re-compiling the plugin for 8.5 - assuming that the user compiles for 8.5 rather than simply bumping the version in the plugin descriptor.

Reply:
It's difficult to decide who these migration docs should be for, the plugin author, or the user installing such a plugin. They aren't necessarily the same person. Here I've focused on the user, so they know nothing of compiling the plugin.